### PR TITLE
add option "BBOX" for returnGeomAs

### PR DESCRIPTION
**Pull request by mdsumner**
New feature for type of geometry return.

A followup from #538

### CODE REVIEW

**Comment by mdsumner** [comment]
I didn't think about the gis attributes: 

```
 $ geom        : num [1:4] 502690 -12896 505713 -10726
 - attr(*, "gis")=List of 5
  ..$ type         : chr "vector"
  ..$ geom_column  : chr "geom"
  ..$ geom_col_type: chr "MULTIPOLYGON"
  ..$ geom_col_srs : chr "PROJCS[\"NAD83 / Montana\",GEOGCS[\"NAD83\",DATUM[\"North_American_Datum_1983\",SPHEROID[\"GRS 1980\",6378137,2"| __truncated__
  ..$ geom_format  : chr "BBOX"
 - attr(*, "class")= chr [1:2] "OGRFeature" "list"
 
```


geom_col_type doesn't really make sense now, just "" ?   And the plot method could use wk::rct, will wait on your first response

**Review by ctoney — COMMENTED** [review]
Thanks! This looks good. I'm neutral on `geom_col_type` for this case. Empty string makes sense for now. The gis attributes might change still so we can revisit whether `geom_col_type` is even useful.

You may know a better way, but I was curious about plotting. I didn't see `as_rct()` for a list of rectangle vectors similar to `as_wkb()` for a list of raw vectors. But a matrix works, so I ended up with this in case it's useful, for a list `x` of bbox vectors:
```r
wk_obj <- matrix(unlist(x), ncol = 4, byrow = TRUE) |> wk::as_rct()
wk::wk_plot(wk_obj)
```

File: src/gdalvector.cpp
Comment (mdsumner) [reply src/gdalvector.cpp:1234]: ok will do, appreciate the explanation - I'm still confused about handle vs pointer, and it didn't work the way I expected at first

File: R/gdalvector.R
Comment (mdsumner) [reply R/gdalvector.R:179]: done

Comment (mdsumner) [reply R/gdalvector.R:179]: eek, I didn't realize I had edited *your* comment above, here's what I'd said

> Oh I did not know you could construct rct that way, that's excellent, usually I pass the four columns in individually to `rct()`.  wk is always column-based afaik so that makes sense. (Oh I see, you can `as_xy()` a matrix  in the same way, otherwise `xy()` is separate columns  ...). rgl has a bunch of "reasonable assumptions" that work like this too and I really like that behaviour.

**Comment by mdsumner** [comment]
folded in the suggestions

File: src/gdalvector.cpp
Comment (ctoney) [inline src/gdalvector.cpp:1237]: Thanks for this code comment. I need to think about that elsewhere.

**Review by ctoney — APPROVED** [review]
All looks good to me. Do you plan to add plotting in this PR? Note that I have #541 pending with commits in both of the S3 plot methods. I can modify that PR to also add BBOX if you like, otherwise if you add plot here I'll rebase after this is merged. Either works for me, please proceed as you see fit. Thanks again!

File: src/gdalvector.cpp
Comment (mdsumner) [reply src/gdalvector.cpp:1237]: oh please do the plotting for this, I can't focus on this for a while now

**Comment by mdsumner** [comment]
oh please do the plotting for this, I can't focus on this for a while now

**Pull request merged**
